typeinf.ml

```diff
@@ -0,0 +1,68 @@
+open Lang
+
+
+exception Error of string 
```

Comment:
Give a meaningful name to the exception please, so that it's immediately clear what breaks when it's thrown.